### ISSUE TITLE
Print out Python versions in compare output

### ISSUE THREAD

**Issue by brettcannon** [python/performance#11]
It would be nice to print out more version details in the compare output to act as a double-check that the results are using data from interpreters that are expected.

**Comment by haypo** [comment]
You can manually inspect JSON files using the perf module directly. Example:

```
python3 -m perf show --metadata test.json
Common metadata:
- aslr: Full randomization
- cpu_affinity: 3,7 (isolated)
- cpu_config: 3,7=driver:intel_pstate, intel_pstate:no turbo, governor:powersave
- cpu_count: 8
- cpu_model_name: Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
- hostname: smithers
- loops: 1
- perf_version: 0.7.8
- performance_version: 0.2.2
- platform: Linux-4.5.7-300.fc24.x86_64-x86_64-with-fedora-24-Twenty_Four
- python_cflags: -Wno-unused-result -Wsign-compare -Wunreachable-code -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv
- python_executable: /home/haypo/prog/python/performance/venv/cpython3.5-19471deca1cb/bin/python
- python_implementation: cpython
- python_version: 3.5.1 (64-bit)
- timer: clock_gettime(CLOCK_MONOTONIC), resolution: 1.00 ns

2to3
----

Metadata:
- cpu_freq: 3=3400 MHz, 7=1942 MHz
- cpu_temp: coretemp:Physical id 0=54 C, coretemp:Core 0=49 C, coretemp:Core 1=50 C, coretemp:Core 2=47 C, coretemp:Core 3=54 C
- date: 2016-09-13T11:28:01
- description: Performance of the Python 2to3 program
- duration: 2.4 sec
- load_avg_1min: 1.46
- mem_max_rss: 13.0 MB
- runnable_threads: 2

Median: 2.38 sec

(...)
```

Sadly, I removed the --metadata parameter from the "perf compare" command. I should reimplement it.

But there are many metadata, the output can be seen as too verbose.

Yeah, we can at least write a short summary of the most important metadata. For example, display python_executable + python_version?

**Comment by brettcannon** [comment]
I think that would be enough.

**Comment by haypo** [comment]
Fixed by the commit 4c566f1595ae8c5d627740aa2ab5c79053cb1839. Python version is now displayed.

By the way, I added a new "show" command. Since the old Python benchmark suite (benchmarks) didn't support writing results into a file, there was no need for such command before.

The display_suite_metadata() function of performance/compare.py is written to be extensible (display more metadata).

I chose to not display the python_executable metadata because we currently get a file like ".../venv/cpython3.5-8d80a6da3c25/bin/python" which is not the original path. Maybe performance should override this metadata somehow? At least for the first run which just created the virtual environment? But I suggest to open a new issue for that, if you consider that it's worth it.
